src/main/java/org/influxdb/impl/InfluxDBResultMapper.java

```diff
@@ -263,9 +324,11 @@ String getMeasurementName(final Class<?> clazz) {
       if (value instanceof String) {
         instant = Instant.from(ISO8601_FORMATTER.parse(String.valueOf(value)));
       } else if (value instanceof Long) {
-        instant = Instant.ofEpochMilli((Long) value);
+        instant = Instant.ofEpochMilli(toMillis((Long) value, precision));
```

Comment:
Could you please do a tinny improvement here by casting to primitives instead of Object wrappers? If you do, then please change the method signature from `toMillis(final Long value, ...` to `toMillis(final long value, ...` (just to avoid unboxing).